File: plantpal/client/lib/models/care_log.dart
class CareLog {
  final int id;
  final int plantId;
  final String action;
  final DateTime date;
  final String? notes;

  CareLog({
    required this.id,
    required this.plantId,
    required this.action,
    required this.date,
    this.notes,
  });

  factory CareLog.fromJson(Map<String, dynamic> json) {
    return CareLog(
      id: json['id'],
      plantId: json['plant_id'],
      action: json['action'],
      date: DateTime.parse(json['date']),
      notes: json['notes'],
    );
  }
}

File: plantpal/server/app/models/care_log.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base
from datetime import datetime

class CareLog(Base):
    __tablename__ = "care_logs"
    id = Column(Integer, primary_key=True, index=True)
    plant_id = Column(Integer, ForeignKey("plants.id"))
    action = Column(String)  # "полив", "удобрение", "пересадка"...
    date = Column(DateTime, default=datetime.utcnow)
    notes = Column(String, nullable=True)
    plant = relationship("Plant", back_populates="care_logs")
